### DISCUSSION TITLE
aws-sdk-cpp for VSCode & GitHub

### DISCUSSION THREAD

**Discussion by NxtGenLegend** [aws/aws-sdk-cpp#3048] [Q&A] [answered]
Hey, I've downloaded the aws-sdk-cpp, and I'm hoping to use that for VSCode, I know that there is an extension for it as well, but my aim here was to use the aws-sdk-cpp & make it work with my project via PATH so that it can be used with any editor.

Additionally, wanted to work on it such that when the repo is cloned, the sdk is downloaded automatically with it, or atleast the person who is cloning the repo only needs to download the sdk and some amount of setup is done already, what would work?

**Comment by jmklix** [comment]
Are you asking for how it can be added to your own project? Because there are multiple options for different use cases. It sounds like you might just want to have the compiled binaries for this SDK put somewhere in the PATH, but I will need more details about your use case to give you a better guidance

**Comment by NxtGenLegend** [comment]
Hey, thank you for responding. So for clarification, I'm looking for the following usecases:
- With VSCode with AWS toolkit
- With a general text editor and/or VSCode without AWS toolkit (with aws-sdk-cpp)
- To manage automatic installation when cloning repo

**Reply by jmklix** [reply]
- I sorry, but I don't think I understand what you mean by "With VSCode with AWS toolkit". What exactly are you trying to use this sdk for in parallel with the AWS toolkit? 
- Same as above
- You can have a shell script to run the necessary cmake commands to build and install this sdk. This would need to be customized for each OS/environment.

**Reply by NxtGenLegend** [reply]
Hey Joseph, thanks for replying again, so for the first usecase, there is no use of the sdk, just the aws toolkit, for the second usecase it is only the sdk (I've installed it, but how do I get to using it in my code now). 

Thank you for solving the third usecase!

**Reply by jmklix** [reply]
Check out our [hello_s3 sample](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/build-cmake.html), which shows the use of the sdk in a standalone c++ program

**Reply by NxtGenLegend** [reply]
Hey yeah I've done that, but is there a way to do it without packaging it as an application? Essentially so I can just run code on VSCode (currently I get an import error, as attached below), from what I know, there is a VSCode extension that works with AWS, so I was hoping to use that, BUT I also want to know how I can resolve this issue without using it so that other text editors can also work with my code
![Issue](https://github.com/user-attachments/assets/ee5cdbbb-e2c8-4384-ae24-3cccd123538b)
[2A152D4A-0166-9AAA-214B8C08A828651A.pdf](https://github.com/user-attachments/files/16371868/2A152D4A-0166-9AAA-214B8C08A828651A.pdf)

**Reply by DmitriyMusatkin** [reply]
C++ is not really a scripting language, so its not something you can "run" within vscode. 
You will need to install C++ development plugin in VS Code.
Then you build C++ AWS SDK as described in the readme and artifacts of that build will be installed in whatever folder you specified in CMAKE_INSTALL_PREFIX.
And finally to get your C++ dev vs code plugin to find the sdk you will have to add whatever path you specified in CMAKE_INSTALL_PREFIX to include path (vscode will typically take you to the setting you need to modify if the c++ plugin is installed correctly)

**Reply by NxtGenLegend** [reply]
Understood, I had the plugin & everything setup, the issue was arising from incorrect PATH variables in VSCode, but that's fixed now. Thank you for your help. I'm running into one final error which is a bit weird, but as shown in the screenshot below, it's attempting to access a config.h header file, however there is only a config.h.in file, and hence it's having trouble accessing, what can I do to fix this?
![image](https://github.com/user-attachments/assets/5ea41806-dae0-44f0-83cf-68b501896a3e)

this is consequently causing a problem in my header file as well

![image](https://github.com/user-attachments/assets/545ef010-cc85-4145-b3f2-244688476d96)

I've added all include folders in my PATH variables already

**Reply by DmitriyMusatkin** [reply, marked as answer]
config.h is a file generated by cmake - https://github.com/awslabs/aws-c-common/blob/main/CMakeLists.txt#L31. and it should be in the generated include folder after you build the sdk.

**Reply by NxtGenLegend** [reply]
That was the issue, thank you so much!
